src/sympc/tensor/mpc_tensor.py

```diff
@@ -688,14 +688,22 @@ def __apply_public_op(
 
         Raises:
             ValueError: If "op_str" is not supported.
+            TypeError: if share_class is not supported.
         """
+        from sympc.tensor import ReplicatedSharedTensor
```

Comment:
Q: If we import this at the top of the module we get an error, right?

Reply:
Yes ,George Circular import error